model/waterfall.go

```diff
@@ -0,0 +1,618 @@
+package model
```

Comment:
we typically use the model package for things that are represented in the DB. 

I think this model should probably all end up in the rest package?

additionally this needs way more tests.

Reply:
Yeah, that's correct. I should be in the API. I'm not sure I can import rest v2 from ui service (don't remember if it lead to issues with dependencies loop).